akka-javasdk/src/main/scala/akka/javasdk/impl/client/AgentClientImpl.scala

```diff
@@ -170,4 +172,72 @@ private[javasdk] final case class AgentClientImpl(
         .asJava
 
   }
+
+  override def dynamicCall[A1, R](agentId: String): ComponentMethodRef1[A1, R] = {
+    import MetadataImpl.toSpi
+
+    val agentClass = agentClassById.getOrElse(
+      agentId,
+      throw new IllegalArgumentException(
+        s"No agent with id [$agentId]. " +
+        "The agent id is defined with the @ComponentId annotation."))
+
+    val effectMethod =
+      agentClass.getDeclaredMethods
+        .find { method =>
+          isCommandHandlerCandidate[Agent.Effect[_]](method) || isCommandHandlerCandidate[Agent.StreamEffect](method)
+        }
+        .getOrElse {
+          // this should have been validated at startup
+          throw new IllegalStateException(s"Agent [${agentClass.getName}] doesn't have expected effect method")
+        }
```

Comment:
@johanandren Does this look like the right way to find the Method? and then create the ComponentMethodRefImpl?

Seems to work in the test I added.

Reply:
Yep, something like this 👍